inference_aws.py

```diff
@@ -56,17 +56,72 @@ def download_directory_from_s3(bucket_name,
 
 download_directory_from_s3('gauganspade', 'datasets_mini')
 
+# need a separate command to download vgg19 weight from S3
+
+
 # add download for attnGAN in WK3
 
-# Part 2-1
+# Part 2
 # AttnGAN
 
 # to be added: a way to programmatically find the directory then save files
 
 print("MS AttnGAN inference")
 
-bashCommand = 'sudo python /home/ubuntu/DeepDeco/src/attngan/code/main.py --cfg ' \
+bashCommand20 = 'sudo python /home/ubuntu/DeepDeco/src/attngan/code/main.py --cfg ' \
               '/home/ubuntu/DeepDeco/src/attngan/code/cfg/eval_coco.yml --gpu 0 '
 print("running bash command")
 
-os.system(bashCommand)
\ No newline at end of file
+os.system(bashCommand20)
+
+# Part 3
+# Deep Lab
+
+print("DeepLab V2 segmentation: setup conda environment")
+
+bashCommand30 = 'conda env create -f configs/conda_env.yaml'
+
+os.system(bashCommand30)
+
+bashCommand31 = 'conda activate deeplab-pytorch'
+
+os.system(bashCommand31)
+
+print("DeepLab V2 segmentation: download caffemodel pre-trained on ImageNet and 91-class COCO (1GB+)")
+
+bashCommand32 = 'sudo bash /home/ubuntu/DeepDeco/src/deeplab/scripts/setup_caffemodels.sh'
+
+os.system(bashCommand32)
+
+print("DeepLab V2 segmentation: Convert the caffemodel to pytorch compatible.")
+
+bashCommand33 = 'sudo python /home/ubuntu/DeepDeco/src/deeplab/convert.py --dataset coco'
+
+os.system(bashCommand33)
+
+print("DeepLab V2 segmentation: now segmenting image....")
+
+bashCommand34 = 'sudo python /home/ubuntu/DeepDeco/src/deeplab/demo.py single' \
+                 ' -c /home/ubuntu/DeepDeco/src/deeplab/configs/coco.yaml' \
+                 ' -m /home/ubuntu/DeepDeco/src/deeplab/data/models/coco/deeplabv1_resnet101/caffemodel' \
+                 '/deeplabv1_resnet101-coco.pth' \
+                 ' -i /home/ubuntu/DeepDeco/0_s_0_g1.png'
+
+os.system(bashCommand34)
+
+print("DeepLab V2 segmentation: completed")
+
+
+# Part 4
+# Deep photo style transfer
+
+print("Deep photo style transfer inference")
+
+bashCommand40 = "sudo python /home/ubuntu/DeepDeco/src/deepphoto/deep_photostyle.py --content_image_path " \
+                "/home/ubuntu/DeepDeco/datasets_mini/coco_stuff/val_img/000000000632.jpg --style_image_path " \
+                "/home/ubuntu/DeepDeco/0_s_0_g1.png --content_seg_path " \
+                "/home/ubuntu/DeepDeco/bed3_inputbb.png --style_seg_path " \
+                "/home/ubuntu/DeepDeco/chair10_tarbb.png --style_option 2 "
+
+os.system(bashCommand40)
+
```

Comment:
This file can potentially use Bash script instead of Python script since you mostly just run bash command.

Reply:
indeed! I will change it this week.